core/src/chain/header.rs

```diff
@@ -131,14 +149,10 @@ impl ToJson for Header {
 //
 impl ToEntry for Header {
     fn to_entry(&self) -> Entry {
-        Entry::new(
-            EntryType::Header.as_str(),
-            &self.to_json().expect("entry should be valid"),
-        )
+        Entry::from(self.to_json().expect("entry should be valid"))
```

Comment:
The `ToEntry` trait needs to change to `ToEntryHeader` trait. Its important to set the entry_type here.